src/Sylius/Bundle/CoreBundle/Controller/ProductTaxonController.php

```diff
@@ -48,45 +50,89 @@ public function updatePositionsAction(Request $request): Response
                 } catch (\InvalidArgumentException $exception) {
                     throw new HttpException(Response::HTTP_BAD_REQUEST, $exception->getMessage());
                 }
-
-                $this->manager->flush();
             }
         }
 
         return new JsonResponse();
     }
 
-    /**
-     * @psalm-suppress DeprecatedMethod
-     */
     public function updateProductTaxonsPositionsAction(Request $request): Response
     {
         $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+        $this->validateCsrfProtection($request, $configuration);
         $this->isGrantedOr403($configuration, ResourceActions::UPDATE);
-        $productTaxons = $this->getParameterFromRequest($request, 'productTaxons');
+        $productTaxonsPositions = $request->request->all('productTaxons');
```

Comment:
Since you've changed it to `::all` the signature of and check within `::shouldProductsPositionsBeUpdated` should be updated. `::all` always returns an array.
The usage of `::getParameterFromRequest` in the method above should also be updated accordingly to the change here.

Reply:
I think it's better now.